sklearn/preprocessing/tests/test_data.py

```diff
@@ -2526,3 +2526,16 @@ def test_standard_scaler_sparse_partial_fit_finite_variance(X_2):
     scaler = StandardScaler(with_mean=False)
     scaler.fit(X_1).partial_fit(X_2)
     assert np.isfinite(scaler.var_[0])
+
+
+def test_minmax_scaler_clip():
+    # test to add a new paramter 'clip' to MinMaxScaler
+    X = iris.data
+    x = np.random.randint(0, 50, (20, 4))
+    scaler = MinMaxScaler(clip=True)
+    # default params, feature_range=(0,1)
+    X_scaled = scaler.fit(X)
+    x_transformed = X_scaled.transform(x)
+
+    assert x_transformed.min() >= 0
+    assert x_transformed.max() <= 1
```

Comment:
This can test can be more explicit:

```py
    X = iris.data
    scaler = MinMaxScaler(clip=True)
    # default params, feature_range=(0,1)
    X_scaled = scaler.fit(X)

    X_min = np.min(X, axis=0)
    X_max = np.max(X, axis=0)

    X_test = [np.r_[X_min[:2], X_max[2:]]]
    X_transformed = X_scaled.transform(X_test)
    assert_allclose(X_transformed, [[0, 0, 1, 1]])
```

Reply:
Thank you for the suggestions :). I have made the changes but I am not sure why the following tests are failing.